File: 14.longest-common-prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        result = ""
        i = 0
        while True:
            if i >= len(strs[0]):
                break
            j = 1
            while j < len(strs):
                if i >= len(strs[j]) or strs[j][i] != strs[0][i]:
                    return result
                j += 1
            result += strs[0][i]
            i += 1
        return result

File: 14.longest-common-prefix/solution.go
func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	i := 0
	for {
		if i >= len(strs[0]) {
			return strs[0][0:i]
		}
		for j := 1; j < len(strs); j++ {
			if i >= len(strs[j]) || strs[j][i] != strs[0][i] {
				return strs[0][0:i]
			}
		}
		i++
	}
	return strs[0]
}
